runtime/src/main/cpp/Worker.cpp

```diff
@@ -475,6 +492,25 @@ class State {
     }
   }
 
+  void waitNativeWorkerTerminationUnlocked(KInt id) {
```

Comment:
Code duplication here is suspicious. Can we avoid it? For example, always take an `id` parameter, with `id == -1` meaning "all workers"? Or, probably more C++-styled solution, take a lambda filtering worker ids.

Reply:
Good idea, thanks!